pageserver/src/tenant.rs

```diff
@@ -1715,6 +1715,9 @@ impl Tenant {
 
     pub fn set_new_tenant_config(&self, new_tenant_conf: TenantConfOpt) {
         *self.tenant_conf.write().unwrap() = new_tenant_conf;
+        for timeline in self.timelines.lock().unwrap().values() {
+            timeline.tenant_conf_updated();
+        }
```

Comment:
Had missed this part earlier, wondering if the timelines should be cloned to a list before letting them know of the config change. Feels like a high risk situation to be calling something towards the tenant and deadlocking later.

Reply:
This would be something like:

```suggestion
        let timelines = self.timelines.lock().unwrap().values().collect::<Vec<_>>();
        for timeline in timelines {
            timeline.tenant_conf_updated();
        }
```

Reply:
```suggestion
        let timelines = self.timelines.lock().unwrap().values().collect::<Vec<_>>();
        // notify the timelines after dropping the lock
        for timeline in timelines {
            timeline.tenant_conf_updated();
        }
```

Good catch!